language/move-core/types/src/parser.rs

```diff
@@ -375,8 +375,50 @@ pub fn parse_struct_tag(s: &str) -> Result<StructTag> {
     }
 }
 
+/// Test cases for the parser that should always fail.
+/// Note: we don't have `cfg(test)` here because code gated by `cfg(test)` is
+/// not gated outside of its declaring crate: <https://github.com/rust-lang/cargo/issues/8379>
+pub const PARSE_VALUE_NEGATIVE_TEST_CASES: &[&str] = &[
```

Comment:
I think it would be better to copy this everywhere it is used rather than exposing this (rather confusing) public constant. It will also make the relevant tests much easier to read/extend.

In general, I think code reuse in production code is great, but we should prioritize readability and avoiding coupling over code reuse in tests. As a rule of thumb, someone touching test code in component A should be able to easily read and understand the tests without looking at test code in component B, and should be able to freely change test inputs/outputs in component A without worrying about changing tests in component B.

Reply:
This was a carry-over from the original PR.
But I agree with you and will change it.